tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts

```diff
@@ -490,176 +490,287 @@ describe('image card', () => {
-        selectSpy,
-        PluginType.IMAGES,
-        'card1',
-        null /* metadataOverride */,
-        timeSeries
-      );
-      const fixture = createImageCardContainer('card1');
-      fixture.detectChanges();
 
-      const dots = fixture.debugElement.queryAll(
-        By.css('.linked-time-ticks-wrapper .linked-time-tick')
-      );
-      expect(dots.length).toBe(2);
-      // The second and third tick is selected.
-      expect(dots[0].nativeElement.getAttribute('style')).toBe(TICKS_STYLE[1]);
-      expect(dots[1].nativeElement.getAttribute('style')).toBe(TICKS_STYLE[2]);
-    });
+      it('does not render ticks on slected range wrapped between steps ', () => {
+        store.overrideSelector(selectors.getMetricsSelectedTime, {
+          start: {step: 11},
+          end: {step: 14},
+        });
 
-    it('renders ticks on steps are particially in range', () => {
-      store.overrideSelector(selectors.getMetricsSelectedTime, {
-        start: {step: 25},
-        end: {step: 350},
+        const fixture = createImageCardContainer('card1');
+        fixture.detectChanges();
+
+        const dots = fixture.debugElement.queryAll(
+          By.css('.linked-time-wrapper .linked-time-tick')
+        );
+        // The third and fourth ticks are selected.
+        expect(dots.length).toBe(0);
       });
-      const timeSeries = [
-        {wallTime: 100, imageId: 'ImageId1', step: 10},
-        {wallTime: 101, imageId: 'ImageId2', step: 20},
-        {wallTime: 102, imageId: 'ImageId3', step: 30},
-        {wallTime: 103, imageId: 'ImageId4', step: 40},
-      ];
-      provideMockCardSeriesData(
-        selectSpy,
-        PluginType.IMAGES,
-        'card1',
-        null /* metadataOverride */,
-        timeSeries
-      );
-      const fixture = createImageCardContainer('card1');
-      fixture.detectChanges();
 
-      const dots = fixture.debugElement.queryAll(
-        By.css('.linked-time-ticks-wrapper .linked-time-tick')
-      );
-      // The third and fourth ticks are selected.
-      expect(dots[0].nativeElement.getAttribute('style')).toBe(TICKS_STYLE[2]);
-      expect(dots[1].nativeElement.getAttribute('style')).toBe(TICKS_STYLE[3]);
+      it('does not render ticks when the slected range is clipped', () => {
+        store.overrideSelector(selectors.getMetricsSelectedTime, {
+          start: {step: 45},
+          end: {step: 55},
+        });
+
+        const fixture = createImageCardContainer('card1');
+        fixture.detectChanges();
+
+        const dots = fixture.debugElement.queryAll(
+          By.css('.linked-time-wrapper .linked-time-tick')
+        );
+        // The third and fourth ticks are selected.
+        expect(dots.length).toBe(0);
+      });
     });
 
-    it('does not render ticks on slected range wrapped between steps ', () => {
-      store.overrideSelector(selectors.getMetricsSelectedTime, {
-        start: {step: 11},
-        end: {step: 14},
+    describe('sliders', () => {
```

Comment:
Could you add a test for start.step being on a tick and end not on a tick and vice versa.